app/src/main/java/com/futurice/freesound/network/api/ApiNetworkModule.java

```
@@ -69,20 +65,12 @@ static FreeSoundApi provideFreeSoundApi(@Named(API_URL_CONFIG) String url,
     @Provides
     @Singleton
     @ForFreeSoundApi
-    static Gson provideGson() {
-        return new GsonBuilder()
-                .setDateFormat(ApiConstants.DATE_FORMAT_PATTERN)
-                .registerTypeAdapterFactory(FreesoundTypeAdapterFactory.create())
-                .create();
-    }
-
-    @Provides
-    @Singleton
-    @ForFreeSoundApi
     static Moshi provideMoshi() {
         return new Moshi.Builder()
                 .add(new KotlinJsonAdapterFactory())
                 .add(new GeoLocationJsonAdapter())
+                .add(new DateAdapter())
```

I've spent like 5h on the issue. If one adapter is not included, we get json parsing error. If other is not included, it fails silently.